static/resources/harness.js

```diff
@@ -764,50 +764,73 @@
     resultsEl.appendChild(resultEl);
   }
 
+  function sendReport(results) {
+    var body = JSON.stringify(results);
+
+    var client;
+    if ('XMLHttpRequest' in self) {
+      client = new XMLHttpRequest();
+    } else if ('ActiveXObject' in self) {
+      client = new ActiveXObject('Microsoft.XMLHTTP');
+    }
+
+    if (!client) {
+      updateStatus(
+        'Cannot upload results: XMLHttpRequest is not supported.',
+        'error-notice'
+      );
+      return;
+    }
+
+    var resultsURL =
+      (location.origin || location.protocol + '//' + location.host) +
+      '/api/results?for=' +
+      encodeURIComponent(location.href);
+
+    client.open('POST', resultsURL);
+    client.setRequestHeader('Content-Type', 'application/json;charset=UTF-8');
+    client.send(body);
+    client.onreadystatechange = function () {
+      if (client.readyState == 4) {
+        if (client.status >= 200 && client.status <= 299) {
+          document.getElementById('export-download').disabled = false;
+          document.getElementById('export-github').disabled = false;
+          updateStatus('Results uploaded.', 'success-notice');
+        } else {
+          updateStatus(
+            'Failed to upload results: server error.',
+            'error-notice'
+          );
+          consoleLog('Server response: ' + client.response);
+        }
+      }
+    };
+  }
+
   function report(results, hideResults) {
     updateStatus('Tests complete. Posting results to server...');
 
     try {
-      var body = JSON.stringify(results);
+      if ('JSON' in self && 'parse' in JSON) {
+        sendReport(results);
+      } else {
+        // Load JSON polyfill if needed
+        var polyfill = document.createElement('script');
+        polyfill.src = '/resources/json3.min.js';
 
-      var client;
-      if ('XMLHttpRequest' in self) {
-        client = new XMLHttpRequest();
-      } else if ('ActiveXObject' in self) {
-        client = new ActiveXObject('Microsoft.XMLHTTP');
-      }
+        if ('onload' in polyfill) {
+          polyfill.onload = function () {
+            sendReport(results);
+          };
+        } else {
+          // If we can't determine when the polyfill loads, use a delay
```

Comment:
If this turns out to be unreliable (due to import taking we could have the imported script call a function we've defined. But only if it becomes a problem in practice.